plugins/utils/refactoring.nix

```diff
@@ -0,0 +1,164 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.refactoring =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "refactoring.nvim";
+
+      package = helpers.mkPackageOption "refactoring.nvim" pkgs.vimPlugins.refactoring-nvim;
+
+      promptFuncReturnType =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrsOf bool
+        )
+        ''
+          {
+            "go" = false;
+            "java" = false;
+            "cpp" = false;
+            "c" = false;
+            "h" = false;
+            "hpp" = false;
+            "cxx" = false;
+          }
+        ''
+        ''
+          For certain languages like Golang, types are required for functions that return an object(s) and parameters of functions.
+          Unfortunately, for some parameters and functions there is no way to automatically find their type. In those instances,
+          we want to provide a way to input a type instead of inserting a placeholder value.
+        '';
+
+      promptFuncParamType =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrsOf bool
+        )
+        ''
+          {
+            "go" = false;
+            "java" = false;
+            "cpp" = false;
+            "c" = false;
+            "h" = false;
+            "hpp" = false;
+            "cxx" = false;
+          }
+        ''
+        ''
+          For certain languages like Golang, types are required for functions that return an object(s) and parameters of functions.
+          Unfortunately, for some parameters and functions there is no way to automatically find their type. In those instances,
+          we want to provide a way to input a type instead of inserting a placeholder value.
+        '';
+
+      printVarStatements =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrOf (listOf str)
```

Comment:
CI fails due to typo: attrOf --> attrsOf
You should run `nix flake check`, it would find those sorts of things

Reply:
Yeh sorry, I just usually like to push my changes. I'll try to check before pushing :sweat: